addons/sale_management/models/sale_order_option.py

```diff
@@ -107,15 +107,17 @@ def _compute_discount(self):
 
     def _get_values_to_add_to_order(self):
         self.ensure_one()
-        return {
+        res = {
             'order_id': self.order_id.id,
             'price_unit': self.price_unit,
             'name': self.name,
             'product_id': self.product_id.id,
             'product_uom_qty': self.quantity,
             'product_uom': self.uom_id.id,
-            'discount': self.discount,
         }
+        if self.discount:
+            res['discount'] = self.discount
+        return res
```

Comment:
Not convinced, I understand this kind of logic for relational fields (to only apply those only if set), but I don't like that for float/monetary fields, what if you put 0 on purpose ?

I prefer to say that pricelist are not supported than saying they are half supported ...

cc: @dalaOdoo

Reply:
The problem is tht with the broken way we compute "partial period" in subscription.
Currently we need to trigger a computation of discount otherwise all newly added lines are at full price as there is no need to compute the "partial period" discount (because a value already exists.

I agree that it's a subscription problem and that we should work on a way to handle upsell in better capacity in subscription but as of now, it's the only "quick" solution I have.

Reply:
If it's only a subscription problem, then override `_get_values_to_add_to_order` in `sale_subscription` (it already depends on `sale_management`.
```suggestion
        if self.is_subscription and self.discount:
            res['discount'] = self.discount
        return res
```